crypto/evp/evp_pbe.c

```diff
@@ -100,36 +105,43 @@ int EVP_PBE_CipherInit(ASN1_OBJECT *pbe_obj, const char *pass, int passlen,
             i2t_ASN1_OBJECT(obj_tmp, sizeof(obj_tmp), pbe_obj);
         ERR_raise_data(ERR_LIB_EVP, EVP_R_UNKNOWN_PBE_ALGORITHM,
                        "TYPE=%s", obj_tmp);
-        return 0;
+        goto err;;
     }
 
     if (pass == NULL)
         passlen = 0;
     else if (passlen == -1)
         passlen = strlen(pass);
 
-    if (cipher_nid == -1)
-        cipher = NULL;
-    else {
-        cipher = EVP_get_cipherbynid(cipher_nid);
```

Comment:
Keep in mind that engines still need to be supported for a while.. If the fetch fails does it need to try legacy for now?